ui-manchette-with-spacetimechart/src/helpers.ts

```diff
@@ -24,37 +24,41 @@ export const calcWaypointsToDisplay = (
 
   // For proportional display, we only display waypoints that do not overlap with
   // the last displayed point:
-  const result: InteractiveWaypoint[] = [{ ...waypoints[0], display: true }];
   const totalDistance = calcTotalDistance(waypoints);
   const heightWithoutFinalWaypoint = getHeightWithoutLastWaypoint(height);
-  let lastDisplayedWaypoint = result[0];
+  const minSpace = BASE_WAYPOINT_HEIGHT / yZoom;
 
   // We iterate through all points, and only add them if they don't collide
   // with the last visible point:
-  for (let i = 1; i < waypoints.length; i++) {
-    const waypoint = waypoints[i];
-    const diff = waypoint.position - lastDisplayedWaypoint.position;
-    const display =
-      (diff / totalDistance) * heightWithoutFinalWaypoint * yZoom >= BASE_WAYPOINT_HEIGHT;
-
-    if (display) {
-      result.push({ ...waypoint, display });
-      lastDisplayedWaypoint = waypoint;
-    }
-  }
+  const sortedWaypointsByWeight = waypoints
+    .slice()
+    .sort((a, b) => (b.weight ?? 0) - (a.weight ?? 0));
```

Comment:
I have this typescript message `The property 'weight' does not exist on type 'Waypoint'.`